common/src/main/java/com/wynntils/wynn/model/PlayerRelationsModel.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.ChatPacketReceivedEvent;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.FriendListUpdateEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import net.minecraft.network.chat.ChatType;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+/**
+ * This model handles the player's in-game relations, like friends, party info, guild info.
+ */
+public class PlayerRelationsModel extends Model {
+    private static final Pattern FRIEND_LIST_MESSAGE_PATTERN = Pattern.compile(".+'s friends \\(.+\\): (.*)");
+    private static final Pattern FRIEND_REMOVE_MESSAGE_PATTERN =
+            Pattern.compile("§e(.+) has been removed from your friends!");
+    private static final Pattern FRIEND_ADD_MESSAGE_PATTERN = Pattern.compile("§e(.+) has been added to your friends!");
+
+    private static boolean expectingFriendMessage = false;
+
+    private static Set<String> friends;
+
+    public static void init() {
+        friends = Set.of();
+
+        if (WorldStateManager.onServer()) {
+            tryGetFriendList();
+        }
+    }
+
+    public static void disable() {
+        friends = Set.of();
+    }
+
+    @SubscribeEvent
+    public static void onChatReceived(ChatPacketReceivedEvent e) {
+        if (e.getType() != ChatType.SYSTEM) return;
+
+        if (expectingFriendMessage) {
```

Comment:
We should probably test this with an account that has no friends...

Reply:
(I know for certain there were a long-standing bug in legacy where we tried to detect guild members and it failed if you were not part of a guild)

Reply:
I happen to have an account for this :)

Reply:
I wonder if we could parse the tab list first, and if that is full only then run the command

Reply:
This is more reliable imo.

Reply:
The tab list cut long names short, so it will not work, unfortunately. :( Iirc, this was done in the packet sent by the server, and not only on client rendering.

Reply:
True

Reply:
Fixed